### PR TITLE
feat(query): support parallel final aggregator

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat(query): support parallel final aggregator

Fixes #issue

### CODE REVIEW

File: src/common/base/src/base/thread_pool.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+use async_channel::Sender;
+use common_exception::Result;
+
+use crate::base::Thread;
+use crate::base::ThreadJoinHandle;
+
+// Simple thread pool implementation,
+// which can run more tasks on limited threads and wait for all tasks to finished.
+pub struct ThreadPool {
+    #[allow(dead_code)]
```
Comment (Xuanwo) [inline src/common/base/src/base/thread_pool.rs:11]: Not used?

File: src/common/base/Cargo.toml [hunk 2]
```diff
@@ -44,6 +44,7 @@ tikv-jemalloc-sys = "0.4.3"
 tokio = { version = "1.21.1", features = ["full"] }
 tracing = "0.1.36"
 uuid = { version = "1.1.2", features = ["serde", "v4"] }
+async-channel = "1.7.1"
```
Comment (Xuanwo) [inline src/common/base/Cargo.toml:47]: I'm guessing we need to run `taplo fmt`

**Comment by Xuanwo** [comment]
```shell
ERROR the following files don't have a valid license header: 
src/common/base/src/base/thread_pool.rs
src/common/base/tests/it/thread_pool.rs
src/query/service/src/pipelines/processors/transforms/aggregator/aggregate_info.rs
src/query/service/src/pipelines/processors/transforms/aggregator/aggregator_final_parallel.rs 
```

**Review by Xuanwo — APPROVED** [review]

File: src/query/service/src/pipelines/processors/transforms/aggregator/aggregator_final_parallel.rs [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+            for (_, bucket_blocks) in std::mem::take(&mut self.buckets_blocks) {
+                data_blocks.extend(bucket_blocks);
+            }
+
+            let method = self.method.clone();
+            let params = self.params.clone();
+            let mut bucket_aggregator = BucketAggregator::<HAS_AGG, _>::create(method, params)?;
+            generate_blocks = bucket_aggregator.merge_blocks(data_blocks)?;
+        } else if self.buckets_blocks.len() > 1 {
+            info!("Merge to final state using a parallel algorithm.");
```
Comment (BohuTANG) [inline src/query/service/src/pipelines/processors/transforms/aggregator/aggregator_final_parallel.rs:116]: A info log in a hot path?

Comment (zhang2014) [reply src/query/service/src/pipelines/processors/transforms/aggregator/aggregator_final_parallel.rs:116]: call once per thread.

**Comment by BohuTANG** [comment]
Test results on 32C32G:

All table counts:
```
+----------+
| count()  |
+----------+
| 64000000 |
+----------+
```


SQL: 
`select sum(uid) from log group by did,sid ignore_result;`
 `group by did, sid` unique key count:`1000000`


```
main branch:
mysql> select sum(uid) from log group by did,sid ignore_result;
Empty set (10.56 sec)
Read 64000000 rows, 5.74 GiB in 10.555 sec., 6.06 million rows/sec., 557.30 MiB/sec.

This PR:
mysql> select sum(uid) from log group by did,sid ignore_result;
Empty set (5.64 sec)
Read 64000000 rows, 5.74 GiB in 5.625 sec., 11.38 million rows/sec., 1.02 GiB/sec.
```

2X faster! 🚀

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**